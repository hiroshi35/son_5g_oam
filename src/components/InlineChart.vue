<template>
  <span class="inline-chart"><span ref="chart" v-show="false"><slot></slot></span></span>
</template>

<script>
  import 'peity'
  export default {
    props: {
      type: {
        type: String,
        default: 'bar'
      },
      options: {
        type: Object,
        default () {
          return {}
        }
      }
    },
    mounted () {
      $(this.$refs.chart).peity(this.type, this.options)
    }
  }
</script>
<style>
  .inline-chart {
    position: relative;
    margin-left: 2px;
    margin-right: 2px;

  }

  .inline-chart svg {
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
  }
</style>
