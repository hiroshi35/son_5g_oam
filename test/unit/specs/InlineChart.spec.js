import { shallow } from 'vue-test-utils'
import InlineChart from '@/components/InlineChart.vue'
import {peity} from 'jquery'

describe('InlineChart.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(InlineChart,
      {
        propsData: {
          type: 'line'
        },
        slots: {
          default: '4.2/5'
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has correct props', () => {
    expect(wrapper.props().type).toBe('line')
  })
  test('$().peity() should be called with line', () => {
    expect(peity).toBeCalled()
    expect(peity.mock.calls[0][0]).toBe('line')
  })
})
