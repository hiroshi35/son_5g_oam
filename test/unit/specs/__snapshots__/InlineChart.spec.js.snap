// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineChart.vue has the expected html structure 1`] = `
<span
  class="inline-chart"
>
  <span
    style="display: none;"
  >
    4.2/5
  </span>
</span>
`;
